api/pkg/apis/v1alpha1/managers/activations/activations-manager.go

```diff
@@ -110,6 +116,16 @@ func (m *ActivationsManager) UpsertState(ctx context.Context, name string, state
 	}
 	state.ObjectMeta.FixNames(name)
 
+	if m.needValidate {
+		if state.ObjectMeta.Labels == nil {
+			state.ObjectMeta.Labels = make(map[string]string)
+		}
+		state.ObjectMeta.Labels["campaign"] = state.Spec.Campaign
```

Comment:
If the activation is updated, will the labels be updated? Or if we guard update in webhook, we need to add the same logic for activiation vendor?

Reply:
activation spec update is always blocked, so labels will not be updated because of activation spec. 
All the crud operation for resources will go through managers so I think validation in the manager should be fine.